perftest/simulations/src/test/scala/org/projectnessie/perftest/gatling/CommitToBranchSimulationDifferentTables.scala

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.perftest.gatling
+
+import io.gatling.core.Predef._
+import io.gatling.core.scenario.Simulation
+import io.gatling.core.structure.{ChainBuilder, ScenarioBuilder}
+import org.projectnessie.client.api.NessieApiV1
+import org.projectnessie.client.http.HttpClientBuilder
+import org.projectnessie.error.NessieConflictException
+import org.projectnessie.model.Operation.Put
+import org.projectnessie.model._
+import org.projectnessie.perftest.gatling.Predef.nessie
+
+import scala.concurrent.duration.{FiniteDuration, HOURS, NANOSECONDS, SECONDS}
+
+/** Gatling simulation to perform commits against Nessie. It commits the data to
+  * different table - each user has its own dedicated table. Has a bunch of
+  * configurables, see the `val`s defined at the top of this class.
+  */
+class CommitToBranchSimulationDifferentTables extends Simulation {
+
+  val params: CommitToBranchParams = CommitToBranchParams.fromSystemProperties()
+
+  /** The actual benchmark code to measure Nessie-commit performance in various
+    * scenarios.
+    */
+  private def commitToBranch: ChainBuilder = {
+    val chain = exec(
+      nessie("Commit")
+        .execute { (client, session) =>
+          // The commit number is the loop-variable declared buildScenario()
+          val commitNum = session("commitNum").asOption[Int].get
+          // Current Nessie Branch object
+          val branch = session("branch").as[Branch]
+          // Our "user ID", an integer supplied by Gatling
+          val userId = session.userId
+          // Table used in the Nessie commit
+          val tableName = params.makeTableName(session)
+
+          // Call the Nessie client operation to perform a commit
+          val key = ContentKey.of("name", "space", tableName, userId.toString)
+          val contentId = tableName + "_" + userId.toString
+
+          val tableMeta = IcebergTable
+            .of(
+              s"path_on_disk_${tableName}_${userId}_$commitNum",
+              42,
+              43,
+              44,
+              45,
+              contentId
+            )
+
+          val existingTable =
```

Comment:
@nastra I don't see how if should remove this state at all. Right now, the state is per table and table is per user so keeping this state is not problematic.